fastqheat.py

```diff
@@ -10,131 +10,106 @@
 from download import download_run_aspc, download_run_fasterq_dump, download_run_ftp
 from study_info.get_sra_study_info import get_run_uid
 
+SRR_PATTERN = re.compile(r'^(SRR|ERR|DRR)\d+$')
+SRP_PATTERN = re.compile(
+    r'^(((SR|ER|DR)[PAXS])|(SAM(N|EA|D))|PRJ(NA|EB|DB)|(GS[EM]))\d+$'
+)
 
-def handle_methods(term, method, out):
-    SRR_pattern = re.compile(r'^(SRR|ERR|DRR)\d+$')
-    SRP_pattern = re.compile(
-        r'^(((SR|ER|DR)[PAXS])|(SAM(N|EA|D))|PRJ(NA|EB|DB)|(GS[EM]))\d+$'
-    )
-    if method == "f":
-        if SRR_pattern.search(term) is not None:
-            accession = term
+
+def _handle_f_method(term, out):
+    if SRR_PATTERN.search(term) is not None:
```

Comment:
`if SRR_PATTERN.search(term):`

Reply:
Functionally equivalent, strongly depends on personal preference. I prefer comparing values with `None` explicitly.

Reply:
Removed explicit `is not None` comparisons.

Reply:
> if search() returns None or any other nullable

`re.search` is documented to  "Return None if no position in the string matches the pattern". It cannot return any other falsy value.

Reply:
So if it cannot return any other nullable values, why check whether it is `None` and not "", [], {}, 0?

If we had code like these:
```python
if foo is None:
    do_this()
elif foo == []:
    do_that()
```
that would make sense

Reply:
Like I said, personal preference.

Reply:
I don't quite understand how it is about personal preferences, to be honest. If it is not necessary to check exactly for None, why check exactly for None? There should be a reason to do or not to do something, imo. We code for money, we refactor code to make it more maintainable, we check for `None` if it makes any difference. If not, why?